### PR TITLE
Completed #ifdef use_deprecated_io

### LINKED ISSUE #1060: mpp_io and fms_io compile time choice
**Is your feature request related to a problem? Please describe.**
Model Systems Division will only provide updates and support for fms2_io.  As a result, we will deprecate mpp_io and fms_io.  These modules and the associated routines and functions that use them in the libFMS code will remain in the code for an extended period of time, but will not be compiled by default.

**Describe the solution you'd like**
mpp_io and fms_io will be surrounded by ifdefs and will not be compiled by default.  
```fortran
module mpp_io_mod
#ifdef use_deprecated_io
...
#endif
end module mpp_io_mod
```
In order to compile the old IO, you will have to use the flag `-Duse_deprecated_io` when compiling libFMS.

**Describe alternatives you've considered**
None

**Additional context**
There is currently no timeline for deleting the mpp_io, fms_io, and the associated routines from the code.  A timeline will be established when current generation GFDL models are all supported using fms2_io.

### PR DESCRIPTION
**Pull request by mcallic2**
**Description**
Adds `#ifdef use_depreciated_io` around all fms_io modules and any calls included in old io 

Fixes #1060 

**How Has This Been Tested?**
Tested with autotools on the sky box with oneapi

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

File: test_fms/parser/parser_demo.F90 [hunk 1]
```diff
@@ -18,6 +18,7 @@
 !***********************************************************************
 
 program parser_demo
+#ifdef use_deprecated_io
```
Comment (rem1776) [inline test_fms/parser/parser_demo.F90:21]: this one shouldn't be ifdef'd, doesn't use any old io.

**Review by rem1776 — CHANGES_REQUESTED** [review]
I left some comments, but I think this will need some other changes as well cause the data override test is failing in the ci.

**Review by uramirez8707 — CHANGES_REQUESTED** [review]
I don't see the `#ifdef use_deprecated_io ` around mpp/mpp_io.F90

File: test_fms/mpp/test_mpp_gatscat.F90 [hunk 2]
```diff
@@ -34,7 +34,9 @@ program test_mpp_gatscat
   use mpp_mod, only : mpp_sync
   use mpp_mod, only : mpp_declare_pelist, mpp_set_current_pelist, mpp_set_stack_size
   use mpp_mod, only : mpp_gather, mpp_scatter, mpp_error, FATAL
-  use mpp_io_mod, only: mpp_io_init, mpp_flush
+#ifdef use_deprecated_io  
+  use mpp_io_mod, only: mpp_flush
```
Comment (rem1776) [inline test_fms/mpp/test_mpp_gatscat.F90:38]: you can delete this use and any calls to mpp flush if there is any

File: test_fms/mpp/test_mpp_sendrecv.F90 [hunk 3]
```diff
@@ -34,7 +34,9 @@ program test_mpp_sendrecv
   use mpp_mod, only : mpp_sync
   use mpp_mod, only : mpp_declare_pelist, mpp_set_current_pelist, mpp_set_stack_size
   use mpp_mod, only : mpp_send, mpp_recv, mpp_error, FATAL
-  use mpp_io_mod, only: mpp_io_init, mpp_flush
+#ifdef use_deprecated_io  
+  use mpp_io_mod, only: mpp_flush
```
Comment (rem1776) [inline test_fms/mpp/test_mpp_sendrecv.F90:38]: same as last comment

**Review by rem1776 — CHANGES_REQUESTED** [review]

File: axis_utils/axis_utils.F90 [hunk 4]
```diff
@@ -27,6 +27,7 @@
 !> @addtogroup axis_utils_mod
 !> @{
 module axis_utils_mod
+#ifdef use_deprecated_io
```
[the remark this answers is not in the document]
Comment (uramirez8707) [reply axis_utils/axis_utils.F90:30]: I think it should be empty, the code will not compile if someone tries to use any of the `axis_utils_mod`

**Comment by uramirez8707** [comment]
The coupler CI is failing because `ocean_null` has not been updated to use fms2_io
```
ocean_null/ocean_model.F90:26:28:

   26 | use           fms_mod, only: field_size, field_exist, get_mosaic_tile_grid, set_domain
      |                            1
Error: Symbol 'field_size' referenced at (1) not found in module 'fms_mod'

```

I updated `ocean_null` a long time ago, but never got into the main repo
https://github.com/NOAA-GFDL/ocean_null/pull/4

**Comment by thomas-robinson** [comment]
@uramirez8707 and @mcallic2 What is being done for fixing the failed coupler compile build test?  Do we know when we can update it so that the build works?

**Comment by uramirez8707** [comment]
I think @nikizadehgfdl needs to merge 
https://github.com/NOAA-GFDL/ocean_null/pull/4 And then re run the CI

File: test_fms/mpp/test_mpp_gatscat.F90 (hunk 2 above)
Comment (rem1776) [reply test_fms/mpp/test_mpp_gatscat.F90:38]: @mcallic2 could you remove these `mpp_flush` statements for this and the other test? I don't think they are even used and it'd be better to not use the old io at all in unrelated tests

File: mpp/mpp_io.F90 [hunk 5]
```diff
@@ -384,7 +384,7 @@ module mpp_io_mod
   public :: mpp_get_dimension_length, mpp_get_axis_bounds
 
   !--- public interface from mpp_io_misc.h ----------------------
-  public :: mpp_io_init, mpp_io_exit, netcdf_err, mpp_flush, mpp_get_maxunits, do_cf_compliance
```
Comment (rem1776) [inline mpp/mpp_io.F90:386]: can you add this back? still needs to be there so its public for other components

**Review by rem1776 — DISMISSED** [review]

**Review by rem1776 — APPROVED** [review]

**Review by thomas-robinson — APPROVED** [review]

**Comment by thomas-robinson** [comment]
@mcallic2 This needs an issue linked to it

**Pull request merged**